### ISSUE TITLE
implementation of detach() is missing for __arm__ / Teensyduino

### ISSUE THREAD

**Issue by mirdej** [PaulStoffregen/PWMServo#3]
### Description

Tried to compile code that uses PWMServo.detach() on Teensy 3.2 and I get a linker error:
"undefined reference to `PWMServo::detach()'"

implementation is missing in PWMServo.cpp

### Steps To Reproduce Problem

Compile code below for Teensy 3.x

### Hardware & Software

Board: Teensy 3.2
Arduino IDE version: 1.8.5
Teensyduino version : 1.45
Operating system & version: MacOS Mojave

### Arduino Sketch

```cpp
#include <PWMServo.h>

PWMServo	myservo;

void setup(){
  myservo.attach(9);
  myservo.detach();
}

void loop(){
}

### Errors or Incorrect Output
"undefined reference to `PWMServo::detach()'"

**Comment by DougMS3** [comment]
I'm having the same problem - I power up the servo, move it, then power it back down.  The problem is when I power it down the servo has enough juice left in its caps to move to an indeterminate (and very undesired) location.  Detach is necessary to avoid this.

Thank you,

**Comment by jusic** [comment]
I ran at the same issue, being accustomed into using `detach` on AVRs but it didn't work. Basically, the `detach` makes the output go 0% duty, so that the output signal is constantly low. On most (cheap) servos this turns the feedback control loop off so that there is no power going to the servo's motor, even if the servos position changes e.g. when turned manually and the servo is otherwise powered.

As a workaround, to achieve this on Teensy and using this library, you can do

```
void detach_servo(int pin)
{
    analogWrite(pin, 0);
}
```

Note that this of course doesn't stop doing a new `PWMServo::write()` after it, which will again make the servo "attached".

Modifiying the library to support this would be fairly easy, but I don't have time to write & test & do a PR right now.

**Comment by BootsSiR** [comment]
@jusic even on AVRs cheap servos will jump on a detach.  I've been struggling figuring out how to get it to stop happening.

**Comment by jusic** [comment]
@BootsSiR It might be that the workaround or the AVR code are not perfect — setting the zero pulse isn't synchronized with the PWM cycle properly. In that case, the microcontroller might output a short pulse once, which the servo's circuitry/firmware detects and then starts to drive the servo to one direction, and then the signal is cut off.

A lot can depend on the servo firmware support as well. Turnigy TS-910's seem to power off mostly quite nicely, but I have observed some occasional jumps with this particular code too. I suspect that it's related to the unsynchronized PWM cycle with my workaround, but haven't done any further investigation to verify this.

**Comment by BootsSiR** [comment]
@jusic yeah I think you're right.  Looking at the adafruit pwn servo library (https://github.com/adafruit/Adafruit_TiCoServo/blob/master/Adafruit_TiCoServo.cpp) which oddly enough doesn't work at for me I see that in their detach method they ensure that the detach doesn't happen mid pulse.  I believe that is what's happening here as well.  I can't say I fully understand that code enough to implement their solution here though :o

**Comment by BootsSiR** [comment]
@jusic I changed the code a bit by monitoring the output compare register to ensure that we don't detach mid pulse.

I'm not going to submit a pull request as I've only tested the change with my particular atemega328 but for anyone else having this problem my fix is here:

https://github.com/BootsSiR/PWMServo/tree/avr-detach-fix

@DougMS3
